patches/tModLoader/Terraria/ModLoader/PlayerHooks.cs

```diff
@@ -76,22 +76,54 @@ private class HookList
 		}
 
 		private static HookList HookSetupStartInventory = AddHook<Action<List<Item>, bool>>(p => p.SetupStartInventory);
-		private static HookList HookSetupStartInventoryOld = AddHook<Action<List<Item>>>(p => p.SetupStartInventory);
 
 		public static IList<Item> SetupStartInventory(Player player, bool mediumcoreDeath = false) {
 			IList<Item> items = new List<Item>();
 			Item item = new Item();
-			item.SetDefaults(ItemID.CopperShortsword);
-			item.Prefix(-1);
-			items.Add(item);
-			item = new Item();
-			item.SetDefaults(ItemID.CopperPickaxe);
-			item.Prefix(-1);
-			items.Add(item);
-			item = new Item();
-			item.SetDefaults(ItemID.CopperAxe);
-			item.Prefix(-1);
-			items.Add(item);
+
+			if (player.difficulty == PlayerDifficultyID.Creative) {
+				item.SetDefaults(ItemID.IronShortsword);
+				item.Prefix(-1);
+				items.Add(item);
+				item = new Item();
+				item.SetDefaults(ItemID.IronPickaxe);
+				item.Prefix(-1);
+				items.Add(item);
+				item = new Item();
+				item.SetDefaults(ItemID.IronAxe);
+				item.Prefix(-1);
+				items.Add(item);
+				item = new Item();
+				item.SetDefaults(ItemID.IronHammer);
+				item.Prefix(-1);
+				items.Add(item);
+				item = new Item();
+				item.SetDefaults(4281); // TODO: ItemID.FinchStaff
+				item.Prefix(-1);
+				items.Add(item);
+				item = new Item();
+				item.SetDefaults(ItemID.Torch);
+				item.stack = 100;
+				items.Add(item);
+				item = new Item();
+				item.SetDefaults(ItemID.Rope);
+				item.stack = 100;
+				items.Add(item);
+			}
+			else {
+				item.SetDefaults(ItemID.CopperShortsword);
+				item.Prefix(-1);
+				items.Add(item);
+				item = new Item();
+				item.SetDefaults(ItemID.CopperPickaxe);
+				item.Prefix(-1);
+				items.Add(item);
+				item = new Item();
+				item.SetDefaults(ItemID.CopperAxe);
+				item.Prefix(-1);
+				items.Add(item);
+			}
```

Comment:
Did you copy this from vanilla? Isn't it nicer to just have the vanilla method populate this list or???

Reply:
the way tML does it is to discard vanilla's inventory setup anyway, and doing it on its own (ie copper tools for non-journey). I'm all for manually converting the array to a list if needed, just thought i shouldn't deviate from the way tML did it before

Reply:
P.S I have no idea why this change ended up in this PR :/

Reply:
`the way tML does it is to discard vanilla's inventory setup anyway`
I'm not sure if we should continue doing that though ,it sounds dumb